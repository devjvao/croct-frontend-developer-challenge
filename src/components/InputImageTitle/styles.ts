import styled from 'styled-components'

export const TitleWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  flex-flow: column;
  font-size: 1.4rem;
  line-height: 180%;
`

export const Title = styled.div`
  display: grid;
  grid-gap: 1.2rem;
  grid-template-columns: auto 1fr;
  place-items: center;
  font-weight: 500;
  color: #495567;
`

export const Subtitle = styled.span`
  font-weight: normal;
  color: #677489;
`
