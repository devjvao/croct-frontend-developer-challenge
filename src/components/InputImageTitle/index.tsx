import * as S from './styles'

interface Props {
  className?: string
}

const InputImageTitle = ({ className }: Props) => (
  <S.TitleWrapper className={className}>
    <S.Title>
      <em className="icon-image" />
      <span role="title">Organization Logo</span>
    </S.Title>
    <S.Subtitle role="subtitle">
      Drop the image here or click to browse
    </S.Subtitle>
  </S.TitleWrapper>
)

export default InputImageTitle
