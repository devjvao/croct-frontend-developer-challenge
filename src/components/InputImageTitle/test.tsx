import { render, screen } from '@testing-library/react'

import InputImageTitle from '.'

describe('<InputImageTitle />', () => {
  it('should render the title', () => {
    const { container } = render(<InputImageTitle />)

    expect(screen.getByRole('title')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the subtitle', () => {
    render(<InputImageTitle />)
    const subtitle = screen.getByRole('subtitle')

    expect(subtitle).toBeInTheDocument()

    expect(subtitle).toMatchSnapshot()
  })
})
