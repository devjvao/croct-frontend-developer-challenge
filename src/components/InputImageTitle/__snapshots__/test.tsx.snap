// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputImageTitle /> should render the subtitle 1`] = `
.c0 {
  font-weight: normal;
  color: #677489;
}

<span
  class="c0"
  role="subtitle"
>
  Drop the image here or click to browse
</span>
`;

exports[`<InputImageTitle /> should render the title 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  font-size: 1.4rem;
  line-height: 180%;
}

.c1 {
  display: grid;
  grid-gap: 1.2rem;
  grid-template-columns: auto 1fr;
  place-items: center;
  font-weight: 500;
  color: #495567;
}

.c2 {
  font-weight: normal;
  color: #677489;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <em
      class="icon-image"
    />
    <span
      role="title"
    >
      Organization Logo
    </span>
  </div>
  <span
    class="c2"
    role="subtitle"
  >
    Drop the image here or click to browse
  </span>
</div>
`;
